echopype/consolidate/api.py

```diff
@@ -55,84 +60,123 @@ def swap_dims_channel_frequency(ds: Union[xr.Dataset, str, pathlib.Path]) -> xr.
+        Currently only implemented for EK60/EK80 sonar models.
+        Cannot be used in tandem with Echodata Platform group angles.
 
     Returns
     -------
-    The input dataset with a `depth` variable (in meters) added
+    The input dataset with a `depth` variable (in meters) added.
+    """
+    # Open Sv dataset
+    ds = open_source(ds, "dataset", {})
 
-    Notes
-    -----
-    Currently this function only scalar inputs of depth_offset and tilt angle.
-    In future expansion we plan to add the following options:
+    # Raise `ValueError` if `echodata` is needed but not passed in
+    if (not echodata) and (use_platform_vertical_offsets or use_platform_angles or use_beam_angles):
+        raise ValueError(
+            "If any of `use_platform_vertical_offsets` or `use_platform_angles` is `True` "
+            + "then `echodata` cannot be `None`."
+        )
 
-    * Allow inputs as xr.DataArray for time-varying variations of these variables
-    * Use data stored in the EchoData object or raw-converted file from which the Sv is derived,
-      specifically `water_level`, `vertical_offtset` and `tilt` in the `Platform` group.
-    """
-    # TODO: add options to use water_depth, vertical_offset, tilt stored in EchoData
-    # # Water level has to come from somewhere
-    # if depth_offset is None:
-    #     if "water_level" in ds:
-    #         depth_offset = ds["water_level"]
-    #     else:
-    #         raise ValueError(
-    #             "water_level not found in dataset and needs to be supplied by the user"
-    #         )
-
-    # # If not vertical needs to have tilt
-    # if not vertical:
-    #     if tilt is None:
-    #         if "tilt" in ds:
-    #             tilt = ds["tilt"]
-    #         else:
-    #             raise ValueError(
-    #                 "tilt not found in dataset and needs to be supplied by the user. "
-    #                 "Required when vertical=False"
-    #             )
-    # else:
-    #     tilt = 0
+    # Raise `NotImplementedError` if `use_platform_angles` and `use_beam_angles` are
+    # both true.
+    if use_platform_angles and use_beam_angles:
+        raise NotImplementedError(
+            "Computing depth with both platform and beam angles is not implemented yet."
+        )
 
-    ds = open_source(ds, "dataset", {})
-    # Multiplication factor depending on if transducers are pointing downward
-    mult = 1 if downward else -1
+    if echodata:
+        # Open Echodata
+        echodata = open_source(echodata, "echodata", {})
 
-    # Compute depth
-    ds["depth"] = mult * ds["echo_range"] * np.cos(tilt / 180 * np.pi) + depth_offset
-    ds["depth"].attrs = {"long_name": "Depth", "standard_name": "depth", "units": "m"}
+        # Grab sonar model
+        sonar_model = echodata["Sonar"].attrs["sonar_model"]
+
+        # Raise value error if sonar model is supported for `use_platform/beam_...` arguments
```

Comment:
In the comment, do you mean "not supported"?

Reply:
Yup I'll make that change 👍